### ISSUE TITLE
[TF2] Checking for updates via parsing server log fails with a syntax error

### ISSUE THREAD

**Issue by athairus** [dgibbs64/linuxgsm#390]
```
tf2server@athair:~$ ./tf2server restart
[  OK  ] Stopping tf2-server: Team Fortress 2
[ .... ] Update tf2-server: Checking for update
[  OK  ] Update tf2-server: Checking for update: Server logsgrep: /home/tf2server/log/console/tf2-server-console.log: No such file or directory

/home/tf2server/functions/fn_update_check: line 77: [: : integer expression expected
[  OK  ] Update tf2-server: Checking for update: Server logs: No update requested
[  OK  ] Update tf2-server: Checking for update: SteamCMD

No update available:
        Current version: 606753
        Available version: 606753
        https://steamdb.info/app/232250/

[  OK  ] Update tf2-server: No update available
[  OK  ] Starting tf2-server: Team Fortress 2
```

When I first ran the script, I believe the file wasn't even there (it's looking for a log without any time/date info in the name?), so I made it:
`touch log/console/tf2-server-console.log`

Its contents as of right now, many many restarts later:
```
tf2server@athair:~$ cat log/console/tf2-server-console.log
Console logging disabled: Bug in tmux 1.8 breaks logging
Currently installed: tmux 1.8
```

This is the command it generated and tried to execute:
`grep -Ec "MasterRequestRestart" "/home/tf2server/log/console/tf2-server-console.log"`

Running it directly in the terminal gives the proper output:
`0`

Something about the context the script executes in must keep it from running properly?

**Comment by dgibbs64** [comment]
Please delete the functions dir and try again. I might of fixed this

**Comment by athairus** [comment]
Just wiped functions/ and tried again. Issue is still there...

**Comment by dgibbs64** [comment]
thanks

**Comment by dgibbs64** [comment]
does the file or directory not exist?

**Comment by athairus** [comment]
It's there, check my OP. Running the command from the terminal works just fine, but the script trips up on it for some reason that must have to do with the environment of the script.

**Comment by dgibbs64** [comment]
Yeah im not sure myself as im unable to replicate. Can you post the output of details (remove passwords etc) so I can test on a similar system.

**Comment by athairus** [comment]
I'll do you one better, send me an RSA public key in the format SSHD recognizes and I'll let you log in and poke around. I'll scrub passwords now.

**Comment by athairus** [comment]
Okay, I can't wait much longer, thought you'd see this message tonight... here's the output from details:

```
tf2server@athair:~$ ./tf2server details

Distro Details
========================================================================================================================================================================================================
Distro:    Ubuntu 14.04.2 LTS
Arch:      x86_64
Kernel:    3.13.0-52-generic
Hostname:  athair.us
tmux:      tmux 1.8
GLIBC:     2.19

Performance
========================================================================================================================================================================================================
Uptime:    0d, 4h, 47m
Avg Load:  0.36, 0.25, 0.20

Mem:       total   used   free
Physical:  994M    922M   71M
Swap:      0B      0B     0B

Disk Usage
========================================================================================================================================================================================================
Disk available:  8.3G
Serverfiles:     14G

Team Fortress 2 Server Details
========================================================================================================================================================================================================
Server name:    CONFIG FILE MISSING!
Server IP:      XXX:27015
RCON password:  CONFIG FILE MISSING!
Status:         ONLINE

Service name:  tf2-server
User:          tf2server
Location:      /home/tf2server
Config file:   /home/tf2server/serverfiles/tf/cfg/tf2-server.cfg CONFIG FILE MISSING!!

Backups
========================================================================================================================================================================================================
No Backups created

Command-line Parameters
========================================================================================================================================================================================================
./srcds_run -game tf -maxplayers 24 -strictportbind -ip XXX -port 27015 +clientport 27005 +exec athairus/sourcetv.cfg +sv_pure 0 +tv_port 27020 +servercfgfile tf2-server.cfg -replay +map pl_badwater

Ports
========================================================================================================================================================================================================
Change ports by editing the command-line
parameters in tf2server.

Useful port diagnostic command:
netstat -atunp | grep srcds_linux

DESCRIPTION  DIRECTION  PORT   PROTOCOL
> Game/RCON  INBOUND    27015  tcp/udp
> SourceTV   INBOUND    27020  udp
< Client     OUTBOUND   27005  udp

Status: ONLINE

```
